src/Raven.Client/Documents/Operations/ETL/AI/AiConnectionString.cs

```diff
@@ -0,0 +1,167 @@
+
+    public OllamaSettings OllamaSettings { get; set; }
+
+    public OnnxSettings OnnxSettings { get; set; }
+
+    public GoogleSettings GoogleSettings { get; set; }
+
+    public HuggingFaceSettings HuggingFaceSettings { get; set; }
+
+    public override ConnectionStringType Type => ConnectionStringType.Ai;
+
+    protected override void ValidateImpl(ref List<string> errors)
+    {
+        var settings = new AbstractAiSettings[]
+        {
+            OpenAiSettings,
+            AzureOpenAiSettings,
+            OllamaSettings,
+            OnnxSettings,
+            GoogleSettings,
+            HuggingFaceSettings
+        }.Where(s => s != null).ToList();
+
+        foreach (var setting in settings)
+        {
+            if (setting.HasSettings() == false)
+                errors.Add($"{setting.GetType().Name} has invalid configuration");
+        }
+
+        if (string.IsNullOrEmpty(Identifier))
+            errors.Add("Could not generate valid identifier from name. Please specify identifier explicitly.");
+
+        switch (settings.Count)
+        {
+            case 0:
+                errors.Add($"At least one of the following settings must be set: {string.Join(", ", nameof(OllamaSettings), nameof(OpenAiSettings), nameof(OnnxSettings))}");
+                break;
+            case > 1:
+                var configuredNames = settings.Select(s => s.GetType().Name);
+                errors.Add($"Only one of the following settings can be set: {string.Join(", ", configuredNames)}");
+                break;
+        }
+    }
+
+    private static string GenerateIdentifier(string input)
+    {
+        if (string.IsNullOrWhiteSpace(input))
+            return null;
+
+        var result = new StringBuilder();
+        var lastWasHyphen = false;
+
+        // First normalize to FormD to separate letters from their diacritics
+        foreach (var c in input.Normalize(NormalizationForm.FormD))
+        {
+            // Check if this is a letter that needs to be preserved
+            if (c is
+                >= 'a' and <= 'z' or
+                >= '0' and <= '9')
+            {
+                result.Append(c);
+                lastWasHyphen = false;
+            }
+            else if (c is >= 'A' and <= 'Z')
+            {
+                result.Append(char.ToLowerInvariant(c));
+                lastWasHyphen = false;
+            }
+            else if (lastWasHyphen == false && result.Length > 0) // Add hyphen for any other character
+            {
+                result.Append('-');
+                lastWasHyphen = true;
+            }
+        }
+
+        // Trim any trailing hyphens
+        var finalResult = result.ToString().TrimEnd('-');
+
+        // Ensure we have at least one character
+        return string.IsNullOrEmpty(finalResult) ? $"{nameof(AiConnectionString)}Identifier" : finalResult;
+    }
+
+    public bool HasCriticalChanges(AiConnectionString newConnectionString)
```

Comment:
as discussed, handle that in a similar way as IndexDefinition changes

Reply:
Done.